### PR TITLE
[Xedra Evolved] Investigation uses, spider infestation et al

### LINKED ISSUE #60790: [Xedra Evolved] Studying spells increases a phantom skill of "nothing"
### Describe the bug

A skill is not set for Xedra Evolved's spells, which generates a phantom skill of "nothing" that does the same function as Magiclysm's spellcraft but cannot be viewed outside of when its level changes.

### Steps to reproduce

1. Spawn in a XE item that teaches a spell, such as ray diagrams.
2. Read until you would level ~~spellcraft~~ whatever XE's equivalent is supposed to be.
3. You learned nothing, literally!

### Expected behavior

~~Not being insulted by the log~~
Levelling a real skill I can view the level of without being in the process of increasing its level.

### Screenshots

image
~While I was reading ray diagrams

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22000.918 (21H2)
- Game Version: 3bfee5a [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Xedra Evolved [xedra_evolved]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Realized from TheShadowFerret's recent issue that I have to include `spellcraft` as a skill in any mod using spells until they are fully jsonized.  I could cheat by leaving id as spellcraft and rename it as deduction but I think I'm fine to wait til spells are fully moddable. Fixes #60790

#### Describe the solution
Adds a chance to treat spider infestations in NPCs, performs some fixes to some things, adds a new monster or two.  Gonna add some additional stuff for a couple dream related traits that exist.

#### Describe alternatives you've considered
None

#### Testing
tbd try to keep from getting so large it's hard to test the whole thing.

### CODE REVIEW

**Comment by Terrorforge** [comment]
When you're doing these PRs, could you please include a slightly more detailed list of what you're adding and how it's supposed to work? I've been trying to review the code, do some testing, and even just play with Xedra stuff in my own game, and it's a bit of a struggle to figure out what is and isn't working as intended, because I don't really know what's intended.

As a pertinent example, I would be testing the vampvirus right now if I had any idea how it was supposed to be acquired and progress through its stages.

**Comment by Maleclypse** [comment]
> When you're doing these PRs, could you please include a slightly more detailed list of what you're adding and how it's supposed to work? I've been trying to review the code, do some testing, and even just play with Xedra stuff in my own game, and it's a bit of a struggle to figure out what is and isn't working as intended, because I don't really know what's intended.
> 
> As a pertinent example, I would be testing the vampvirus right now if I had any idea how it was supposed to be acquired and progress through its stages.


Sure thing. I'll try and put together some additional details in the readme tonight.

File: data/mods/Xedra_Evolved/README.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+### Agents of Change
+I want to take this space to thank GuardianDll for all the time we've talked about what to do in this mod and the work they have put into creating content for this mod.  I'd also like to thank TheShadowFerret for patiently reviewing my PRs no matter how hare brained my sentence structure is at any given time.
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/README.md:2]:
```suggestion
I want to take this space to thank GuardianDll for all the time we've talked about what to do in this mod and the work they have put into creating content for this mod.  I'd also like to thank TheShadowFerret for patiently reviewing my PRs no matter how hare brained my sentence structure is at any given time.  Ramza13 for creating the EOC system that I've used so heavily in this mod and all of the dev team for their assistance in correcting my attempts and sometimes even building c++ functions related to all the mods and systems I work on.
```

File: data/mods/Xedra_Evolved/modinfo.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
...
+    "description": "A content mod for Agent Mulder in the apocalypse.  Creates a multi decade timeline of dimensional spillage prior to the cataclysm.  This spillage was characterized by thin places appearing where people could step through into another dimension but things from those dimensions could wander into our world as well.  Xedra proliferated into an alphabet soup of agencies dedicated to simultaneously stopping, controlling, and utilizing these thin places with varying degrees of succes.  Contains monsters of various types, new professions and locations, new uses for existing skills plus the new deduction skill, new magic systems, and content of every type that a multidimensional Cataclysm has to offer.  This is a heavily science fantasy mod where you are as likely to encounter supernatural monsters, as you are to encounter UFOs and Sasquatch and Chupacabra.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/modinfo.json:8]:
```suggestion
    "description": "A content mod for Agent Mulder in the apocalypse.  Creates a multi decade timeline of dimensional spillage prior to the cataclysm.  This spillage was characterized by thin places appearing where people could step through into another dimension but things from those dimensions could wander into our world as well.  Xedra proliferated into an alphabet soup of agencies dedicated to simultaneously stopping, controlling, and utilizing these thin places with varying degrees of success.  Contains monsters of various types, new professions and locations, new uses for existing skills plus the new deduction skill, new magic systems, and content of every type that a multidimensional Cataclysm has to offer.  This is a heavily science fantasy mod where you are as likely to encounter supernatural monsters, as you are to encounter UFOs and Sasquatch and Chupacabra.  Please examine the readme.md file for further details.",
```

File: data/mods/Xedra_Evolved/npc/rubik.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  {
+    "id": "TALK_EXODII_MERCHANT_VAMPIRE_GEAR",
+    "type": "talk_topic",
+    "dynamic_line": "The Good Folk of Aberystwyth?  Us 'ave no truck with them, the milk is spoilt, the flesh is hoofshod, the henge holds no peel.  Ye'd ken best not their seductions.  A trifle better than the enemy.",
+    "//": "The faeries announced themselves in Rubik's world near what we know as Wales.  Exodii mostly avoid them referencing several myths about bad luck and that for because they've traded their flesh for metal the fae aren't really interested in them anyway.  Not quite as bad as the undead but more tricksome.",
+    "responses": [
+      { "text": "I'll keep your warnings in mind.", "topic": "TALK_EXODII_MERCHANT_Talk" }
+    ]
+  }
+]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/rubik.json:57]:
```suggestion
]

```

**Comment by MrDraMaster** [comment]
I made a comment on #60790 (https://github.com/CleverRaven/Cataclysm-DDA/issues/60790#issuecomment-1423280513) that the "nothing" issue seems to have been fixed already by something, so maybe that

**Comment by Maleclypse** [comment]
> I made a comment on #60790 ([#60790 (comment)](https://github.com/CleverRaven/Cataclysm-DDA/issues/60790#issuecomment-1423280513)) that the "nothing" issue seems to have been fixed already by something, so maybe that skills.json bit can be cut?

OK I'll come back to this once 0.G releases.  Thanks!

File: data/mods/Xedra_Evolved/monsters/exodii.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    "bodytype": "horse",
+    "material": [ "steel" ],
+    "speed": 110,
+    "attack_cost": 80,
+    "symbol": "J",
+    "aggression": -10,
+    "morale": 70,
+    "aggro_character": false,
+    "hp": 290,
+    "color": "dark_gray",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/exodii.json:21]:
```suggestion
    "color": "dark_gray",
    "weakpoint_sets": [ "wps_combat_cyborg" ],
    "families": [ "prof_wp_cyborg" ],
```

File: data/mods/Xedra_Evolved/monsters/exodii.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
+[
+  {
+    "id": "mon_exodii_quad_jotunn",
+    "type": "MONSTER",
+    "name": { "str": "Exodii Jotunn" },
+    "//": "Needs a heavy duty weakpoint set that is mostly hardpoints.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/exodii.json:6]:
```suggestion
```

**Pull request merged**